### ISSUE TITLE
Support storing time information with `Ink` annotations

### ISSUE THREAD

**Issue by crlf0710** [pdf-association/pdf-issues#375]
Some applications might want to recover information from the `Ink` annotations. Currently the information stored within a ink annotation dictionary (Table 185) only includes coordinates data, without any temporal data. Time delta between coordinates might be used to infer the "pen" moving speed, and improve the accuracy of the information recovery. Maybe add an optional dictionary entry that supplies such information?

**Comment by petervwyatt** [comment]
There are already two existing mechanisms defined:
* **Metadata** (14.3.2) for XMP-based metadata
* Associated Files (14.13.9) for any arbitrary embedded stream of data

**Comment by petervwyatt** [comment]
So Associated Files could easily be used to store pen pressure, movement, angle, etc. in a 3rd party format.

**Comment by crlf0710** [comment]
Yes, I totally understand that Associated Files can be used to supply "supplement" information. However in this case, it seems temporal information is one of the centric defining property, and it is [distinguishing property between handwriting and casual shapes](https://github.com/WICG/handwriting-recognition/blob/3665f6d40c5a59f93535c042450cb94b46f03fac/explainer.md#why-not-use-shape-detection). So i think the "time" information itself is more important than the "pen pressure, angle, etc" that should be classified as supplement information (supplied by maybe InkML when absolutely needed). In fact, when the `Path` entry is missing and needed to be rebuild from `InkList` with heuristics, if the sampling rate is not uniform, different curves might get generated.

There's indeed reasons about why sometimes "time" should NOT be included (e.g. [for protecting user privacy](https://github.com/WICG/handwriting-recognition/blob/3665f6d40c5a59f93535c042450cb94b46f03fac/explainer.md#why-optional-time)), but i think there's still value in providing such an optional entry.

**Comment by petervwyatt** [comment]
Embedded files allow separate encryption (**EFF**, **AuthEvent EFOpen**, etc) so user privacy can be respected. 
You could also use marked content sequences on the vectors used to make up the path.

**Comment by lrosenthol** [comment]
It is worth noting that there was a proposal (2017/2018?) to TC 171/WG 8 to extend Ink annotations with time, pressure, etc.  Unfortunately, there was not enough interest in the committee to move it forward.  If you (@crlf0710 ) were interested in bring that (or something similar) back, I am sure that ISO committee - or the PDF Association - would welcome your participation in our working groups.
